commit c06308c8154225c795dc1294a9d49794af89f28c
Author: Chris J Arges <christopherarges@gmail.com>
Date:   Fri Mar 17 00:48:19 2017 +0000

    igb: Workaround for igb i210 firmware issue
    
    [ Upstream commit 4e684f59d760a2c7c716bb60190783546e2d08a1 ]
    
    Sometimes firmware may not properly initialize I347AT4_PAGE_SELECT causing
    the probe of an igb i210 NIC to fail. This patch adds an addition zeroing
    of this register during igb_get_phy_id to workaround this issue.
    
    Thanks for Jochen Henneberg for the idea and original patch.
    
    Signed-off-by: Chris J Arges <christopherarges@gmail.com>
    Tested-by: Aaron Brown <aaron.f.brown@intel.com>
    Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
    Signed-off-by: Sasha Levin <alexander.levin@verizon.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linux-4.9.25/drivers/net/ethernet/intel/igb/e1000_phy.c
===================================================================
--- linux-4.9.25.orig/drivers/net/ethernet/intel/igb/e1000_phy.c
+++ linux-4.9.25/drivers/net/ethernet/intel/igb/e1000_phy.c
@@ -81,6 +81,10 @@ s32 igb_get_phy_id(struct e1000_hw *hw)
 	if ((hw->mac.type == e1000_i210) || (hw->mac.type == e1000_i211))
 		phy->ops.write_reg(hw, I347AT4_PAGE_SELECT, 0);
 
+	/* ensure PHY page selection to fix misconfigured i210 */
+	if (hw->mac.type == e1000_i210)
+		phy->ops.write_reg(hw, I347AT4_PAGE_SELECT, 0);
+
 	ret_val = phy->ops.read_reg(hw, PHY_ID1, &phy_id);
 	if (ret_val)
 		goto out;
Index: linux-4.9.25/dummy/rpi_1414_c06308c8154225c795dc1294a9d49794af89f28c.txt
===================================================================
--- /dev/null
+++ linux-4.9.25/dummy/rpi_1414_c06308c8154225c795dc1294a9d49794af89f28c.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
